### DISCUSSION TITLE
DefaultValues Warning Motivation

### DISCUSSION THREAD

**Discussion by bbrinx** [react-hook-form/react-hook-form#2916] [General]
First of all, thank you a bunch for this library, so far it's been very pleasant to use! I would like to bring up an issue I've been having lately to see if there's something I'm missing.

After updating to the latest react-hook-form version I am seeing warning messages for not providing default values for all form fields. What's the motivation behind adding warning messages for this? What are the downsides of not providing a default value for a field? 

From my perspective this adds a lot of noise. There are two main scenarios I can think of in which a form doesn't need default values passed into the `useForm` hook:

1) Form has no query data and therefore does not need default values. For example a file upload form. I now have to add a typed defaultValues object, setting every field to null.
2) Form has query data, which is loaded async and will be set using `reset`. I now have to create a defaultValues object defining every field as null, just for it to be overwritten once the data is loaded. 

It doesn't seem necessary set the defaultValues in either of these cases. And yet we will get a warning message for every field of that form. 

Of course there might be a good reason for the library to enforce that behavior which I am not aware of. Or I am misunderstanding how to use defaultValues. But as of now, having to add default values to every form has decreased my developer experience with react-hook-form.

**Comment by bluebill1049** [comment]
hey, @bbrinx thanks a lot for your feedback. I think I can summary this and explain to you the rational behind.

as you know RHF is based on uncontrolled forms, such as `<input defaultValue="" />` is for the very first render's input value, this rule applies to controller as well, which defaultValue defined which value is going to render for the input which it's wrapped, now it's "fine" to now include defaultValue, but we have sepefic logic to return a defaultValue or value for the controller, it may not cause a problem for you at this stage but it may, and hence it's a warning not a error. Now you can argume this is completely not necessary for your usage, but at the same time i have to cate for all usage and users, so it's better to give users a warning then see bugs at their product.

**Reply by SergeyWebDeveloper** [reply]
@bluebill1049 , can you give the user the ability to disable this behavior? Not very nice to see a lot of warnings in the console)

**Reply by bluebill1049** [reply]
that would be difficult...
